packages/langium/src/grammar/langium-grammar-validator.ts

```diff
@@ -97,13 +97,15 @@ export class LangiumGrammarValidator {
 
     checkEntryGrammarRule(grammar: ast.Grammar, accept: ValidationAcceptor): void {
         const entryRules = grammar.rules.filter(e => ast.isParserRule(e) && e.entry) as ast.ParserRule[];
-        if (entryRules.length === 0) {
+        if (grammar.isDeclared && entryRules.length === 0) {
             const possibleEntryRule = grammar.rules.find(e => ast.isParserRule(e) && !isDataTypeRule(e));
             if (possibleEntryRule) {
                 accept('error', 'The grammar is missing an entry parser rule. This rule can be an entry one.', { node: possibleEntryRule, property: 'name', code: IssueCodes.EntryRuleTokenSyntax });
             } else {
                 accept('error', 'This grammar is missing an entry parser rule.', { node: grammar, property: 'name' });
             }
+        } else if(!grammar.isDeclared && entryRules.length === 1) {
+            accept('error', 'A rule marked as entry is illegal if the containing grammar is not declared.', { node: entryRules[0], property: 'name' });
```

Comment:
Minor: This should be performed for every entry rule and run in addition to the other entry rule validation (currently is it running exclusively).

Additionally, minor wording improvement:

```suggestion
            accept('error', 'Cannot declare entry rules for unnamed grammars.', { node: entryRules[0], property: 'name' });
```